### ISSUE TITLE
Missing focus indicators for dashboard

### ISSUE THREAD

**Issue by filiptronicek** [gitpod-io/gitpod#16723]
### Bug description

From https://www.a11yproject.com/posts/never-remove-css-outlines/:
> Removing outlines in CSS creates issues for people navigating the web with a keyboard.

For many elements in the Dashboard, there is no way to tell it is focused without taking action on it. This is bad for folks who use the keyboard to move around.

This is an incomplete list of the components I have encountered with this behavior:
- https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/components/SelectableCardSolid.tsx 
- https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/components/CheckBox.tsx
- https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/components/Button.tsx, specifically type `danger.secondary` has very low contrast, even more pronounced in dark mode.

### Steps to reproduce

v
